### PR TITLE
Add comparsion operator for SQL

### PR DESCRIPTION
**Pull request by penghuo**
## Description of changes:*
1. Add Comparsion operator support for SQL.
https://github.com/penghuo/sql/blob/pr-operator-enhancement/docs/user/dql/expressions.rst#comparison-operators
2. Refactor the BinaryPredicateOperator and UnaryPredicateOperator.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/user/index.rst [hunk 1]
```diff
@@ -22,6 +22,8 @@ Open Distro for Elasticsearch SQL enables you to extract insights out of Elastic
   - `Identifiers <general/identifiers.rst>`_
   - `Data Types <general/datatype.rst>`_
 
+  - `Data Types <general/datatype.rst>`_
```
Comment (dai-chen) [inline docs/user/index.rst:25]: Duplicate?

File: docs/user/dql/expressions.rst [hunk 2]
```diff
@@ -92,6 +92,83 @@ Here is an example for different type of arithmetic expressions::
...
+| IS MISSING     | MISSING value test             |
++----------------+--------------------------------+
+| IS NOT MISSING | NOT MISSING value test         |
++----------------+--------------------------------+
+
+
+Basic Comparison Operator
+-------------------------
+
+Here is an example for different type of comparison operators::::
```
Comment (dai-chen) [inline docs/user/dql/expressions.rst:136]: np: should be `::` here and below?

File: core/src/test/java/com/amazon/opendistroforelasticsearch/sql/expression/operator/predicate/BinaryPredicateOperatorTest.java [hunk 3]
```diff
@@ -458,46 +458,46 @@ public void test_less_null() {
     FunctionExpression less = dsl.less(DSL.literal(1),
         DSL.ref(INT_TYPE_NULL_VALUE_FIELD, INTEGER));
     assertEquals(BOOLEAN, less.type());
-    assertEquals(LITERAL_NULL, less.valueOf(valueEnv()));
+    assertEquals(LITERAL_FALSE, less.valueOf(valueEnv()));
```
Comment (dai-chen) [inline core/src/test/java/com/amazon/opendistroforelasticsearch/sql/expression/operator/predicate/BinaryPredicateOperatorTest.java:461]: So `1 < NULL` returns `FALSE` now?

File: docs/user/index.rst (hunk 1 above)
Comment (penghuo) [reply docs/user/index.rst:25]: Done.

File: docs/user/dql/expressions.rst (hunk 2 above)
Comment (penghuo) [reply docs/user/dql/expressions.rst:136]: Done.

File: core/src/test/java/com/amazon/opendistroforelasticsearch/sql/expression/operator/predicate/BinaryPredicateOperatorTest.java (hunk 3 above)
Comment (penghuo) [reply core/src/test/java/com/amazon/opendistroforelasticsearch/sql/expression/operator/predicate/BinaryPredicateOperatorTest.java:461]: This is the default order in our system.
NULL equal to NULL
NULL large than MISSING
NULL less than all others.

In Order By, the user can also specify the order of NULL and MISSING.

Comment (penghuo) [reply core/src/test/java/com/amazon/opendistroforelasticsearch/sql/expression/operator/predicate/BinaryPredicateOperatorTest.java]: Got your point, change the logic, follow the MySQL way now.
https://dev.mysql.com/doc/refman/8.0/en/working-with-null.html

**Pull request merged**